silverback/_click_ext.py

```diff
@@ -30,6 +34,66 @@ def cls_import_callback(ctx, param, cls_name):
     raise click.BadParameter(message=f"Failed to import {param} class: '{cls_name}'.")
 
 
+def contract_callback(
+    ctx: click.Context, param: click.Parameter, contract_address: str
+) -> ContractInstance:
+    return Contract(convert(contract_address, AddressType))
+
+
+def token_amount_callback(
+    ctx: click.Context,
+    param: click.Parameter,
+    token_amount: str | None,
+) -> int | None:
+    if token_amount is None:
+        return None
+
+    return convert(token_amount, int)
+
+
+def timedelta_callback(
+    ctx: click.Context, param: click.Parameter, timestamp_or_str: str | None
+) -> timedelta | None:
+    if timestamp_or_str is None:
+        return None
+
+    try:
+        timestamp = datetime.fromisoformat(timestamp_or_str)
+    except ValueError:
+        timestamp = None
+
+    if timestamp:
+        if timestamp <= (now := datetime.now()):
+            raise click.BadParameter("Must be a time in the future.", ctx=ctx, param=param)
+        return timestamp - now
+
+    elif " " in timestamp_or_str:
+        units_value = {}
+        for time_units in map(lambda s: s.strip(), timestamp_or_str.split(",")):
+
+            time, units = time_units.split(" ")
+            if not units.endswith("s"):
+                units += "s"
+
+            if units not in {"seconds", "minutes", "hours", "days", "weeks"}:
```

Comment:
(Not blocking) Do we want to include {'s', 'm', 'h', 'd', 'w'} in there? Those are pretty standard for time denominated values with cli tools.

Reply:
Probably needs a lot more work to get something like that to function. Was trying to avoid that right now